filetests/simple_preopt/simplify.clif

```diff
@@ -278,3 +278,16 @@ ebb0:
 ; nextln:     v2 = sextend.i64 v3
 ; nextln:     return v2
 ; nextln: }
+
+function %add_imm_fold(i32) -> i32 {
+ebb0(v0: i32):
+  v1 = iadd_imm v0, 42
+  v2 = iadd_imm v1, -42
+  return v2
+}
+; sameln: function %add_imm_fold(i32)
+; nextln: ebb0(v0: i32):
+; nextln:    v2 -> v0
```

Comment:
currently cranelift turns `add_imm_fold` into:
```
ebb(v0: i32):
  v1 = iadd_imm v0, 42
  v2 -> v1
  nop
  return v2
```

which is definitely not what the function should compute :grin: